### PR TITLE
Update charges on some junkfoods and breads.  Randomize spawns

### PR DESCRIPTION
**Pull request by Maleclypse**
#### Summary

SUMMARY: Content "Update junkfood and bread"

#### Purpose of change
Add variety to spawns.  Also make a single bag of tortilla chips potentially have enough chips to make nachos.

#### Describe the solution

Adds charges where needed to junkfoods and updates them where appropriate, in breads. Randomize the SUS bread distribution.

#### Describe alternatives you've considered
Add snippets describing how the cataclysm came like a thief in the night but strangely every person in the world only had two slices of bread left.

#### Testing
It works!

### CODE REVIEW

**Comment by Maleclypse** [comment]
Oh knock on recipe effects.  I'll deal with you tomorrow.

**Comment by Maleclypse** [comment]
@anothersimulacrum before I go too far down this path.  Have I misunderstood how charges work?  Because I noticed sandwich sauce now has a calorie error and I didn't edit anything that should have broken any sandwiches unless "calories" is divided by "charges" in the item entry itself.

File: data/json/recipes/recipe_food.json [hunk 1]
```diff
@@ -1398,6 +1398,7 @@
     "skill_used": "cooking",
     "difficulty": 3,
     "time": "16 m",
+    "charges":"3",
```
Comment (actual-nh) [inline data/json/recipes/recipe_food.json:1401]:
```suggestion
    "charges": 3,
```

**Comment by anothersimulacrum** [comment]
> @anothersimulacrum before I go too far down this path. Have I misunderstood how charges work? Because I noticed sandwich sauce now has a calorie error and I didn't edit anything that should have broken any sandwiches unless "calories" is divided by "charges" in the item entry itself.

I think what's happening is that the recipe is now producing more (or less) for the same ingredients.

**Comment by Maleclypse** [comment]
> I think what's happening is that the recipe is now producing more (or less) for the same ingredients.

That would make sense but I'm not changing the any related recipes to that sandwich.  The existing recipe has it's own set of charges that I'm not editing. So it shouldn't even be affecting things if charges on the item side dictated how many charges appear in recipes that don't list a separate charges.

**Comment by actual-nh** [comment]
> > I think what's happening is that the recipe is now producing more (or less) for the same ingredients.
> 
> That would make sense but I'm not changing the any related recipes to that sandwich. The existing recipe has it's own set of charges that I'm not editing. So it shouldn't even be affecting things if charges on the item side dictated how many charges appear in recipes that don't list a separate charges.

That does appear to be the case (item charges change result charges if the recipe doesn't specifically list charges); the sandwich_sauce recipe (data/json/recipes/recipes_food.json) does not list its own charges, so I'm not sure what recipe you're meaning.

**Comment by actual-nh** [comment]
The "default calories" the test is claiming is equal to bread_calories (84) times 14 times 2 (sandwich_sauce proportional to bread is *2). In the test:
```
default_calories *= res_it.charges;
```

**Comment by Maleclypse** [comment]
> The "default calories" the test is claiming is equal to bread_calories (84) times 14 times 2 (not sure where that would be from; sandwich_sauce proportional to bread is *2 - see items/comestibles/sandwich.json). In the test:
> 
> ```
> default_calories *= res_it.charges;
> ```

And it shouldn't do that based on my changes I'm thinking. The test is just using the wrong information?  The nachos calories test I absolutely need to make changes to fix.  But I'm thinking for the sauce sandwich I should fix the test and I have no idea how to do that part.

**Comment by actual-nh** [comment]
> And it shouldn't do that based on my changes I'm thinking. The test is just using the wrong information? The nachos calories test is right, I absolutely need to make changes to json entries to fix. But I'm thinking for the sauce sandwich I should fix the test and I have no idea how to do that part.

Could you try modifying the sandwich_sauce recipe in data/json/recipes/recipes_food.json - just add `"charges": 1,`. It may not be supposed to make any difference, but that doesn't mean it won't (and in any event the recipe should have the number of charges produced!). Otherwise - @KorGgenT?

**Comment by Maleclypse** [comment]
> > And it shouldn't do that based on my changes I'm thinking. The test is just using the wrong information? The nachos calories test is right, I absolutely need to make changes to json entries to fix. But I'm thinking for the sauce sandwich I should fix the test and I have no idea how to do that part.
> 
> Could you try modifying the sandwich_sauce recipe in data/json/recipes/recipes_food.json - just add `"charges": 1,`. It may not be supposed to make any difference, but that doesn't mean it won't (and in any event the recipe should have the number of charges produced!). If it doesn't make a difference, then that's information; if it does, then that's both good for this PR and useful information for further fixing. Regarding the test itself 

Looks like you were right and adding the charges to sauce sandwich fixed that result so now I just need to fix the nachos recipe.  Thanks!

**Pull request merged**